### PR TITLE
Fix `structural_transformation` for optimization systems

### PR DESCRIPTION
**Pull request by YingboMa**
Fix https://github.com/SciML/Optimization.jl/pull/539

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -845,13 +845,14 @@ end
 
 # Symbolics needs to call unwrap on the substitution rules, but most of the time
 # we don't want to do that in MTK.
-function fast_substitute(eq::Equation, subs)
-    fast_substitute(eq.lhs, subs) ~ fast_substitute(eq.rhs, subs)
+const Eq = Union{Equation, Inequality}
+function fast_substitute(eq::T, subs) where {T <: Eq}
+    T(fast_substitute(eq.lhs, subs), fast_substitute(eq.rhs, subs))
```
Comment (Vaibhavdixit02) [inline src/utils.jl:850]: This needs the `relational_op` argument as well in case of `Inequality`

**Comment by Vaibhavdixit02** [comment]
I was able to use this branch (after making the change for `Inequality` in `fast_subsititutes`) on a MTK version https://github.com/Vaibhavdixit02/rosetta-opf/blob/main/mtk.jl of the opf problem

File: src/utils.jl (hunk 1 above)
Comment (YingboMa) [reply src/utils.jl:850]: Can you suggest the change?

Comment (YingboMa) [reply src/utils.jl:850]: Also, why should we have `relational_op`? Shouldn't everything normalize to `≲` and define `a ≳ b` as `b ≲ a` due to reflexivity?

Comment (YingboMa) [reply src/utils.jl:850]: Are there any reasons to not enforce it? At least, we can normalize it in OptimizationProblem to save a lot of hassle.

Comment (Vaibhavdixit02) [reply src/utils.jl:850]: No, I don't see any reason not to do it

Comment (YingboMa) [reply src/utils.jl:850]: Cool. I will merge this PR if the tests pass. Do you mind to work on the normalization PR?

Comment (Vaibhavdixit02) [reply src/utils.jl]: So after looking through the constraints code a bit better it looks like we already convert to canonical form https://github.com/SciML/ModelingToolkit.jl/blob/e1a9a06ca1fd26679b792771ecf04e7c11dbe927/src/systems/optimization/constraints_system.jl#L110 and https://github.com/SciML/ModelingToolkit.jl/blob/e1a9a06ca1fd26679b792771ecf04e7c11dbe927/src/systems/optimization/constraints_system.jl#L180-L181 so I don't think we need to it

Comment (Vaibhavdixit02) [reply src/utils.jl]: What can `subs` be other than `Pair`?

Comment (YingboMa) [reply src/utils.jl]: Dict or Vector of pairs

Comment (YingboMa) [reply src/utils.jl]: OK, so undo that commit?

**Pull request merged**